src/fitting.py

```diff
@@ -198,24 +198,30 @@ def validation_cross_entropy(data_val_xyz, labels, model_params, L_max, only_eve
     assert data_val_xyz.shape[1] == 3
     data_val_sph = cart_to_spherical(data_val_xyz)
 
-    unique_labels = np.unique(labels)
-    assert  set(list(unique_labels)).issubset(set(range(len(model_params))))
-    assert unique_labels.shape[0] == len(model_params) or unique_labels.shape[0] == len(model_params) + 1, \
-                f"unique_labels.shape = {unique_labels.shape} and len(model_params) = {len(model_params)}"
-    unique_labels = list(range(len(model_params)))
+    unique_labels = np.unique(labels).astype(int)
 
     # make qs
     qs = []
     for label in unique_labels:
+        if label == -1:
```

Comment:
Is this the bug you mentioned?

Reply:
yeah, what can happen is that all the labels are in a single cluster or the first couple of clusters, one cluster won't have any points in it and the fitting method will through an error. That error tended to happen when using PCA and whittening for some reason